### ISSUE TITLE
Errors when using Axios 1.x.x

### ISSUE THREAD

**Issue by dbelob** [zhaosiyang/axios-observable#35]
Thank you very much for the excellent library!

Errors occurred while migrating to _Axios_ 1.2.1 with axios-observable 2.0.0

**Stackblitz Case (Bug Reports)**
https://stackblitz.com/github/dbelob/axios-observable-cancel-token-error

**Current behavior**
Error when trying to make HTTP request and error message in console.

In _Firefox_:
![image](https://user-images.githubusercontent.com/22131253/207046220-544a7a17-c57d-4f95-838a-1791c6f4579e.png)

In _Chrome_ and _Edge_:
![image](https://user-images.githubusercontent.com/22131253/207046387-e0ad3a27-046f-41dd-9dff-1f7793681755.png)

In _Stackblitz_:
![image](https://user-images.githubusercontent.com/22131253/207046460-c0ddd251-048a-4bc0-ba59-8310bac4f876.png)

**Expected behavior**
Successful completion of the HTTP request, displaying a message in the browser, no error messages in the console:
![image](https://user-images.githubusercontent.com/22131253/207047369-d97cc9d8-cb49-49fd-a8ed-56bfbf9fe1cf.png)
See example with _Axios_ 0.27.2 https://stackblitz.com/github/dbelob/axios-observable-cancel-token-error/tree/axios-0.27.2

* **axios-observable version:** 2.0.0

* **Axios version:** 1.2.1

* **Node.js version:** 18.12.1

* **OS version:** Windows 8.1, Windows 10

* **Browser:** Firefox 107.0.1, Chrome 108.0.5359.99, Edge 108.0.1462.46

**Comment by kernapprentice** [comment]
@dbelob Thank you so much for reporting this issue. I'll be working on it this weekend and keep you updated.

**Comment by zhaosiyang** [comment]
@dbelob Thanks for your feedback.
I can't reproduce the issue that you mentioned. The Stackblitz example just doesn't build.

**Comment by dbelob** [comment]
> @dbelob Thanks for your feedback. I can't reproduce the issue that you mentioned. The Stackblitz example just doesn't build.

Yes, example with **axios-observable 2.0.0** and **Axios 1.2.1** doesn't build on _Stackblitz_:
https://stackblitz.com/github/dbelob/axios-observable-cancel-token-error
(see GitHub repository [dbelob/axios-observable-cancel-token-error](https://github.com/dbelob/axios-observable-cancel-token-error), branch _main_)
![image](https://user-images.githubusercontent.com/22131253/208254854-b5ea3c64-d9b0-48ce-ab6f-09acaba9a046.png)

But the same example with **axios-observable 2.0.0** and **Axios 0.27.2** builds and runs successfully on _Stackblitz_:
https://stackblitz.com/github/dbelob/axios-observable-cancel-token-error/tree/axios-0.27.2
(see GitHub repository [dbelob/axios-observable-cancel-token-error](https://github.com/dbelob/axios-observable-cancel-token-error), branch _axios-0.27.2_)
![image](https://user-images.githubusercontent.com/22131253/208255104-fe860f35-4143-414b-80c3-9c0ecaaed1bc.png)

Summary:
- example with **axios-observable 2.0.0** and **Axios 0.27.2**
  - builds and runs successfully on _Stackblitz_
  - builds and runs successfully in all browsers (_Firefox_, _Chrome_, _Edge_) on the local computer
- example with **axios-observable 2.0.0** and **Axios 1.2.1**
  - doesn't build on _Stackblitz_
  - builds but runs with console errors in all browsers (_Firefox_, _Chrome_, _Edge_) on the local computer

Please do these steps on your local computer:
1. `git clone https://github.com/dbelob/axios-observable-cancel-token-error.git`
2. `cd axios-observable-cancel-token-error`
3. `npm install`
4. `npm start`

(the browser will open automatically with errors in the console)

**Comment by zhaosiyang** [comment]
@dbelob This issue is introduced by `axios@1.1.3` and it works fine for `axios@1.1.2`. Let me try to investigate further. Something might have to be changed on `axios` side. In the mean time, you can revert your `axios` version to 1.1.2 and I'll submit an issue in `axios` once I figure out what's going wrong with `axios`

**Comment by dbelob** [comment]
@zhaosiyang you have closed an unsolved issue.

Repository https://github.com/dbelob/axios-observable-cancel-token-error changed: _Axios_ updated to latest version (1.7.7)

I checked the example with **axios-observable 2.0.0** and **Axios 1.7.7**:
1. `git clone https://github.com/dbelob/axios-observable-cancel-token-error.git`
2. `cd axios-observable-cancel-token-error`
3. `npm install`
4. `npm start`

The browser will open automatically with errors in the console:

In _Firefox_:
![image](https://github.com/user-attachments/assets/cd0eb452-8856-4af3-aeb6-98aa647b631a)

In _Chrome_ and _Edge_:
![image](https://github.com/user-attachments/assets/bdcbd8a9-26e1-42d9-86cb-e264927bc075)

Same errors.

**Comment by antonpegov** [comment]
If someone stragling to figure out "axios_1.default.* is not a function" - I was able to fix it by installing version 1.5.0 of react-observable
